pkgs/development/cuda-modules/cudnn-frontend/default.nix

```diff
@@ -0,0 +1,134 @@
+{
+  autoAddDriverRunpath,
+  catch2_3,
+  cmake,
+  cudaPackages,
+  fetchFromGitHub,
+  gitUpdater,
+  lib,
+  ninja,
+  nlohmann_json,
+  stdenv,
+}:
+let
+  inherit (lib.lists) optionals;
+  inherit (lib.strings)
+    cmakeBool
+    cmakeFeature
+    optionalString
+    ;
+  inherit (cudaPackages)
+    cuda_cccl
+    cuda_cudart
+    cuda_nvcc
+    cuda_nvrtc
+    cudnn
+    libcublas
+    ;
+in
+
+# TODO(@connorbaker): This should be a hybrid C++/Python package.
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cudnn-frontend";
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "cudnn-frontend";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-hKqIWGxVco1qkKxDZjc+pUisIcYJwFjZobJZg1WgDvY=";
+  };
+
+  # TODO: As a header-only library, we should make sure we have an `include` directory or similar which is not a
+  # superset of the `out` (`bin`) or `dev` outputs (whih is what the multiple-outputs setup hook does by default).
+  outputs =
+    [
+      "out"
+    ]
+    ++ optionals finalAttrs.doCheck [
+      "legacy_samples"
+      "samples"
+      "tests"
+    ];
+
+  nativeBuildInputs = [
+    autoAddDriverRunpath # Needed for samples because it links against CUDA::cuda_driver
+    cmake
+    cuda_cccl
+    cuda_nvcc
+    ninja
+  ];
+
+  buildInputs = [
+    cuda_cudart
+  ];
+
+  postPatch =
+    # nlohmann_json should be the only vendored dependency.
+    ''
+      echo "patching source to use nlohmann_json from nixpkgs"
+      rm -rf include/cudnn_frontend/thirdparty/nlohmann
+      rmdir include/cudnn_frontend/thirdparty
+      substituteInPlace include/cudnn_frontend_utils.h \
+        --replace-fail \
+          '#include "cudnn_frontend/thirdparty/nlohmann/json.hpp"' \
+          '#include <nlohmann/json.hpp>'
```

Comment:
Note for future: perhaps we could patch upstream's `target_include_directories`&c stanza make the latter work even when they vendor `thirdparty`?

Reply:
Ideally yes, and it'd be configurable in CMake.